### PR TITLE
Fix: Add warn JSX text literals

### PR DESCRIPTION
**Pull request by interactivellama**

### CODE REVIEW

File: src/scripts/lint/.eslintrc.react.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ module.exports = {
     '@typescript-eslint/no-empty-function': ['error', { allow: ['arrowFunctions'] }],
     'react/jsx-filename-extension': ['error', { extensions: ['.tsx'] }],
     'react/jsx-props-no-spreading': 'off',
+    // Promotes localization of string literals in JSX
+    'react/jsx-no-literals': ['error', { noStrings: false }],
```
Comment (nanovazquez) [inline src/scripts/lint/.eslintrc.react.js:35]: Shouldn't this rule be part of our app? I don't see this being honored in other projects

Comment (interactivellama) [reply src/scripts/lint/.eslintrc.react.js:35]: I don't know the scope of this project and it's goals. Folks were talking about adding Storybook composition with SLDS for React this morning to this project. That seems rather specific, too.

Comment (interactivellama) [reply src/scripts/lint/.eslintrc.react.js:35]: I thought we were supposed to add as many lint rules as possible over here. Who else is using this?

Comment (interactivellama) [reply src/scripts/lint/.eslintrc.react.js:35]: We are not supposed to be writing any strings `{'Hello'}` or `Hello`.

**Pull request merged**